### PR TITLE
Explain double-click behaviour on tool tip

### PR DESCRIPTION
**Pull request by jeenuv**
The outline toggle button has a feature where it can be double-clicked
to expand/collapse all items shown therein. Although this is described
in the FAQ, can go potentially unnoticed. This, however, being a useful
feature, advertise on the tool tip itself.

l10n translations are untouched.

Signed-off-by: Jeenu Viswambharan jeenuv@gmail.com

### CODE REVIEW

**Comment by timvandermeij** [comment]
The localization logic will override this, so this does not work. Instead, you need to edit https://github.com/mozilla/pdf.js/blob/master/l10n/en-US/viewer.properties#L102 and change that to `Show Document Outline (double-click to expand/collapse all)` and use this same string at https://github.com/mozilla/pdf.js/pull/7717/commits/89780640d8f46aa18153639500b29bdb68b9a842#diff-a72ef320c3225c98b2163b89a3e8ff62R84 as well.

**Comment by timvandermeij** [comment]
We might need to change the L10n ID to make the localization teams aware of the change, but I'm not sure if that is actally required.

/cc @yurydelendik @Snuffleupagus What do you think?

**Comment by jeenuv** [comment]
@timvandermeij That's what I did on a local version of this patch, but then I read in [contribution guide](https://github.com/mozilla/pdf.js/wiki/Contributing):

> Note that the translations for PDF.js in the l10n folder are synchronized with the Aurora branch of Mozilla Firefox. This means that we will only accept pull requests that add strings currently missing in the Aurora branch (as it will take at least six weeks before the most recent translations are in the Aurora branch), but keep in mind that the changes will be overwritten when we synchronize again.

My reading was that the patch won't be accepted as it'd be overwritten anyway. I'm happy to push that version instead. What do you think?

**Comment by Snuffleupagus** [comment]
> We might need to change the L10n ID to make the localization teams aware of the change, but I'm not sure if that is actually required.

As explained in https://developer.mozilla.org/en-US/docs/Mozilla/Localization/Localization_content_best_practices#Changing_existing_strings, you ***must always*** update the l10n `id` when changing string, since otherwise the tools used by localizers may not pick up the change!

> [...] and change that to `Show Document Outline (double-click to expand/collapse all)`

Provided that we actually want to change the tooltip unconditionally like this PR suggest, I'm not really sure about the wording used here.
It seems to me that there's something missing after `all`, perhaps it should be `[...] all items)`, since it doesn't feel entirely clear what's actually being expanded/collapsed otherwise (e.g. is it the outline itself?).

**Comment by timvandermeij** [comment]
I agree with the `[...] all items` change.

@jeenuv Everything except for the `en-US` file will be synchronized. We only make changes in the English file and upstream takes care of the other translations that we periodically import.

**Comment by jeenuv** [comment]
I've updated the PR. Please check.

**Comment by timvandermeij** [comment]
This looks better, but we need to change `outline.title` and `outline_label` to `document_outline.title` and `document_outline_label`, respectively. This ID must change to make sure that the localization teams know that they have to update the translation of this string. You need to change this in the `en-US` locale file and in `viewer.html` at the `data-l10n-id` locations.

File: l10n/en-US/viewer.properties [hunk 1]
```diff
@@ -99,8 +99,8 @@ print_progress_close=Cancel
 # tooltips)
 toggle_sidebar.title=Toggle Sidebar
 toggle_sidebar_label=Toggle Sidebar
-outline.title=Show Document Outline
-outline_label=Document Outline
+document_outline_label=Show Document Outline (double-click to expand/collapse all items)
```
Comment (timvandermeij) [inline l10n/en-US/viewer.properties]: The label and title IDs must be swapped, i.e., this should be the title, just like before.

File: web/viewer.html [hunk 2]
```diff
@@ -81,8 +81,8 @@
             <button id="viewThumbnail" class="toolbarButton group toggled" title="Show Thumbnails" tabindex="2" data-l10n-id="thumbs">
                <span data-l10n-id="thumbs_label">Thumbnails</span>
             </button>
-            <button id="viewOutline" class="toolbarButton group" title="Show Document Outline" tabindex="3" data-l10n-id="outline">
-               <span data-l10n-id="outline_label">Document Outline</span>
+            <button id="viewOutline" class="toolbarButton group" title="Show Document Outline (double-click to expand/collapse all items)" tabindex="3" data-l10n-id="outline">
```
Comment (timvandermeij) [inline web/viewer.html]: The `data-l10n-id` in this line should also be `document_outline`.

File: web/viewer.html [hunk 3]
```diff
@@ -81,8 +81,8 @@
...
+               <span data-l10n-id="document_outline.title">Document Outline</span>
```
Comment (timvandermeij) [inline web/viewer.html]: The label ID must be used here, just like before, so this should be `document_outline_label`.

File: web/viewer.html [hunk 4]
```diff
@@ -81,8 +81,8 @@
             <button id="viewThumbnail" class="toolbarButton group toggled" title="Show Thumbnails" tabindex="2" data-l10n-id="thumbs">
                <span data-l10n-id="thumbs_label">Thumbnails</span>
             </button>
-            <button id="viewOutline" class="toolbarButton group" title="Show Document Outline" tabindex="3" data-l10n-id="outline">
-               <span data-l10n-id="outline_label">Document Outline</span>
+            <button id="viewOutline" class="toolbarButton group" title="Show Document Outline (double-click to expand/collapse all items)" tabindex="3" data-l10n-id="document_outline">
+                <span data-l10n-id="document_outline_label">Outline</span>
```
Comment (timvandermeij) [inline web/viewer.html]: Revert the changes on this line except for the `data-l10n-id` change. The extra space is not necessary as we indent everything with two spaces and the text should remain the same as in the L10n file.

**Review by timvandermeij — COMMENTED** [review]
Looks good to me after this last change.

File: web/viewer.html (hunk 4 above)
Comment (jeenuv) [inline web/viewer.html]: Ah, sorry - that must be my editor!

Comment (timvandermeij) [inline web/viewer.html]: The word `Document` has disappeared in this line, so that needs to be put back. Therefore, this line should be:

`<span data-l10n-id="document_outline_label">Document Outline</span>`

**Comment by jeenuv** [comment]
Boy, can't I make a proper commit?! Hopefully I get this one right. Thanks for your patience.

**Review by timvandermeij — APPROVED** [review]
Looks good to me. Since this is a change that affects upstream work for translations, could you merge this if you like it as well, @yurydelendik?

**Review by yurydelendik — APPROVED** [review]

**Comment by timvandermeij** [comment]
Thank you!
